keanu-project/src/main/java/io/improbable/keanu/tensor/dbl/ScalarDoubleTensor.java

```diff
@@ -329,8 +329,8 @@ public DoubleTensor max(DoubleTensor max) {
     }
 
     @Override
-    public DoubleTensor inverse() {
-        return duplicate();
+    public DoubleTensor matrixInverse() {
+        return new ScalarDoubleTensor(1 / value);
```

Comment:
Good find! We need some better scalar testing...

Reply:
Yep - I intend to add some in as part of my testing - it's definitely our weakest area by far at the moment.  Potentially a crazy idea - but should we disable the use of the Scalar objects until we have better testing there?